views/ahnentafel/ahnentafel.js

```diff
@@ -0,0 +1,265 @@
+            this.displayGeneration(nextPeople);
+        }
+    }
+
+    // Grab a parent profile for the next generation. If we don't have an id, we use a place-holder instead, so the 
+    // display keeps going with an "Unknown" relative displayed.
+    async nextPerson(id) {
+        if (id > 0) {
+            let data = await WikiTreeAPI.postToAPI({ 'action': 'getPerson', 'key': id, 'fields': this.profileFields });
+            if (data.length != 1) {
+                return this.blankPerson;
+            } else {
+                return data[0].person;
+            }
+        }
+        return this.blankPerson;
+    }
+
+    // This code takes the WikiTree API person data and renders it into HTML for appending to the view display container.
+    displayPerson(person) {
+        let html = `<p class="ahnentafelPerson" id="person_${this.ahnentafelNumber}">${this.ahnentafelNumber}. `;
+
+        if (person.Id == 0) {
+            html += `[${this.unknownName()} Unknown]`;
+        }
+        else {
+            if (this.generation == 1) {
+                html += '<b>';
+                html += `${person.FirstName} ${person.MiddleName} `;
+                if (person.Nicknames) { html += `${person.NickNames} `; }
+                if (person.LastNameCurrent != person.LastNameAtBirth) { html += ` (${person.LastNameAtBirth}) `; }
+                html += `${person.LastNameCurrent}`;
+                if (person.Suffix) { html += ` ${person.Suffix}`; }
+                html += '</b>';
+            } else {
+                html += `<a name="${this.ahnentafelNumber}"></a>`;
+                html += `<a href="https://www.wikitree.com/wiki/${person.Name}">`;
+                html += `${person.FirstName} ${person.MiddleName} `;
+                if (person.Nicknames) { html += `${person.NickNames} `; }
+                if (person.LastNameCurrent != person.LastNameAtBirth) { html += ` (${person.LastNameAtBirth}) `; }
+                html += `${person.LastNameCurrent}`;
+                if (person.Suffix) { html += ` ${person.Suffix}`; }
+                html += '</a>';
+            }
+            html += ': ';
+
+            if (person.DataStatus.BirthDate != 'blank' || person.DataStatus.BirthLocation != 'blank') { html += 'Born '; }
+            if (person.BirthLocation && person.DataStatus.BirthLocation != 'blank') { html += ` ${person.BirthLocation}`; }
+            if (person.BirthDate != '' && person.BirthDate != '0000-00-00' && person.DataStatus.BirthDate != 'blank' ) {
+                if (person.DataStatus.BirthDate == 'guess') { html += ' about '; }
+                if (person.DataStatus.BirthDate == 'before') { html += ' before '; }
+                if (person.DataStatus.BirthDate == 'after') { html += ' after '; }
+                html += ` ${this.formatDate(person.BirthDate)}`;
+            } else {
+                if (person.DataStatus.BirthDate != 'blank') { html += ' [date unknown]'; }
+            }
```

Comment:
I'm just curious here, is it necessary to check it like this?

`person.DataStatus.BirthDate != 'blank'`

Wouldn't it be enough to check just the field itself?

`person.DataStatus.BirthDate`

Im not really sure how it resolve 🤔. Or does this really resolve as `blank` string? 🤔😱

Btw, beware that `==` is not the same as `===` in javascript.

 https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness